TweenController/TweenPromise.swift

```diff
@@ -36,43 +36,43 @@ public struct TweenPromise<T:Tweenable> {
     let resolvedDescriptors: [TweenDescriptor<T>]
     weak var registration: DescriptorRegistration?
     
-    public func to(_ to: T, at progress: Double, withEasing easing: Easing.Function = Easing.linear) -> TweenPromise<T> {
+    public func to(_ to: T, at progress: Double, withEasing easing: @escaping Easing.Function = Easing.linear) -> TweenPromise<T> {
         assert(progress != self.progress, "'to' progress must be different than 'from' progress")
         
         let descriptor = TweenDescriptor(fromValue: from, toValue: to, interval: self.progress..<progress, easingFunction: easing)
-        registration?.register(descriptor: descriptor)
+        registration?.register(descriptor)
         return TweenPromise(from: to, progress: progress, resolvedDescriptors: resolvedDescriptors + [descriptor], registration: registration)
     }
     
-    public func then(to: T, at progress: Double, withEasing easing: Easing.Function = Easing.linear) -> TweenPromise<T> {
+    public func then(_ to: T, at progress: Double, withEasing easing: @escaping Easing.Function = Easing.linear) -> TweenPromise<T> {
         return self.to(to, at: progress, withEasing: easing)
     }
     
-    public func thenHold(until: Double) -> TweenPromise<T> {
+    public func thenHold(_ until: Double) -> TweenPromise<T> {
         return self.to(from, at: until)
     }
     
-    public func with(action: (T) -> ()) {
+    public func with(_ action: @escaping (T) -> ()) {
```

Comment:
As stated above, the argument labels are helpful in communicating the meaning of the function. We should keep them.